core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/stopstartds/StopStartDownscaleFlowService.java

```diff
@@ -57,18 +58,34 @@ public class StopStartDownscaleFlowService {
     @Inject
     private InstanceMetaDataService instanceMetaDataService;
 
-    public void clusterDownscaleStarted(long stackId, String hostGroupName, Set<Long> privateIds) {
+    public void initScaleDown(long stackId, String hostGroupName) {
         clusterService.updateClusterStatusByStackId(stackId, DetailedStackStatus.DOWNSCALE_BY_STOP_IN_PROGRESS);
         flowMessageService.fireEventAndLog(stackId, Status.UPDATE_IN_PROGRESS.name(), CLUSTER_SCALING_STOPSTART_DOWNSCALE_INIT, hostGroupName);
+    }
 
+    public void clusterDownscaleStarted(long stackId, String hostGroupName, Set<Long> privateIds, Set<String> recoveryCandidateHostIds) {
         // TODO CB-15153: Change the message once an adjustment based downscale is supported.
         LOGGER.debug("stopstart scaling Decommissioning from group: {}, privateIds: [{}]", hostGroupName, privateIds);
         Stack stack = stackService.getByIdWithListsInTransaction(stackId);
         List<String> decomissionedHostNames = stackService.getHostNamesForPrivateIds(stack.getInstanceMetaDataAsList(), privateIds);
-        LOGGER.debug("stopstart scaling Decommissioning from group: {}, hostnames: [{}]", hostGroupName, decomissionedHostNames);
-        flowMessageService.fireInstanceGroupEventAndLog(stackId, Status.UPDATE_IN_PROGRESS.name(), hostGroupName,
-                CLUSTER_SCALING_STOPSTART_DOWNSCALE_STARTING,
-                String.valueOf(decomissionedHostNames.size()), hostGroupName, String.join(", ", decomissionedHostNames));
+        List<String> recoveryCandidateHostNames;
+        if (!recoveryCandidateHostIds.isEmpty()) {
+            recoveryCandidateHostNames = instanceMetaDataService.findByStackIdAndInstanceIds(stackId, recoveryCandidateHostIds)
```

Comment:
please use the instances of the stack, you can filter out from there

Reply:
Used instances from Stack